### DISCUSSION TITLE
More fun with U^2-Net (background removal): web app version with opyrator

### DISCUSSION THREAD

**Discussion by helena-intel** [openvinotoolkit/openvino_notebooks#59] [Show and tell]
Make a standalone web app (works on mobile too) for the [background removal](https://github.com/openvinotoolkit/openvino_notebooks/tree/main/notebooks/205-vision-background-removal) model with [Opyrator](https://github.com/ml-tooling/opyrator)!

Steps:
1) Run the background removal notebook with all three model configuration options, to generate the IR models for Human Segmentation, Salient Object Detection and Salient Object Detection - Lite. Change the model to use by pointing `u2net_model` in the last line in the **Settings** cell to a different model configuration. For example, change `u2net_model = u2net_lite` to `u2net_model = u2net_human_seg` to generate IR models for Human Segmentation. Make sure to run the entire notebook with these settings (choose: Kernel->Restart Kernal and Run All Cells). If all models have been created, the *saved_models* directory (in the *205-vision-background-removal* directory) will contain three subdirectories/folders.
2) Install opyrator with `pip install opyrator`. Make sure to do this after activating `openvino_env`.
3) Download the opyrator app script from https://gist.github.com/helena-intel/a33905119d71e9a63ece9da3e475b4a0 and put it in the 205-vision-background-removal directory. 
4) Launch the app with `opyrator launch-ui segmentation_app:remove_background -p 7777`  (execute this command in a terminal in the 205-vision-background-removal directory).
5) Open the address that is displayed in a web browser. If two addresses are displayed, try the "Network URL". 

It looks like this in a desktop web browser: 
![image](https://user-images.githubusercontent.com/77325899/116816847-f8d8c580-ab63-11eb-862d-425d0f245d05.png)

If you have good or fun results that you would like to share, please do!

Opyrator can do more than turning a function into a web app. See their documentation for more details: https://github.com/ml-tooling/opyrator

Disclaimer: this is meant as a fun thing to try. The app script does not contain any input checking and doesn't catch any exceptions. Do not use this as-is in production or on a public webserver.

**Comment by raymondlo84** [comment]
Omg Meme generator!! Make some examples, and put that on reddit?

**Reply by raymondlo84** [reply]
Will this run properly with a video?

**Reply by helena-intel** [reply]
Not as is, but the code could be adapted to support that. The model is not very fast on CPU though.

**Comment by helena-intel** [comment]
I added an optional caption to meme-ify it. Making the caption look pretty is left as an excercise for the reader :-)
![image](https://user-images.githubusercontent.com/77325899/116876474-d2c32c00-ac1c-11eb-9f1c-5f2dca2488e9.png)

**Reply by ryanloney** [reply]
Nice!!

**Comment by raymondlo84** [comment]
wow cool!
